### PR TITLE
Do not omit nested additionalProperties

### LINKED ISSUE #1497: Additional properties are not submitted when nested and omit data enabled
### Prerequisites

- [ x] I have read the [documentation](https://react-jsonschema-form.readthedocs.io/);
- [ x] In the case of a bug report, I understand that providing a [SSCCE](http://sscce.org/) example is tremendously useful to the maintainers.
- [ x] Ideally, I'm providing a [sample JSFiddle](https://jsfiddle.net/n1k0/f2y3fq7L/6/) or a [shared playground link](https://rjsf-team.github.io/react-jsonschema-form/) demonstrating the issue.

### Description

Additional properties are not submitted when nested and omit data enabled. This only happens when it is nested.

### Steps to Reproduce

1. Goto [Playground example](https://rjsf-team.github.io/react-jsonschema-form/#eyJmb3JtRGF0YSI6eyJuZXN0ZWQiOnsibmV3S2V5IjoiTmV3IFZhbHVlIn19LCJzY2hlbWEiOnsidGl0bGUiOiJXcmFwcGVyIiwiZGVzY3JpcHRpb24iOiJBbiBleGFtcGxlIG9mIE5PVCB3b3JraW5nIGFkZGl0aW9uYWxQcm9wZXJ0aWVzIHdoZW4gbmVzdGVkIiwidHlwZSI6Im9iamVjdCIsInByb3BlcnRpZXMiOnsibmVzdGVkIjp7InR5cGUiOiJvYmplY3QiLCJ0aXRsZSI6Ik5lc3RlZCIsImRlc2NyaXB0aW9uIjoiRW5hYmxlIE9taXQgZXh0cmEgZGF0YSwgYWRkIGEgcmVjb3JkIGFuZCBjbGljayBzdWJtaXQuLi4gc3VibWl0dGVkIGZvcm1EYXRhIGluIGNvbnNvbGUgaXMgZW1wdHkhIiwiYWRkaXRpb25hbFByb3BlcnRpZXMiOnRydWV9fX0sInVpU2NoZW1hIjp7InRhc2tzIjp7Iml0ZW1zIjp7ImRldGFpbHMiOnsidWk6d2lkZ2V0IjoidGV4dGFyZWEifX19fSwibGl2ZVNldHRpbmdzIjp7InZhbGlkYXRlIjp0cnVlLCJkaXNhYmxlIjpmYWxzZSwib21pdEV4dHJhRGF0YSI6dHJ1ZSwibGl2ZU9taXQiOmZhbHNlfX0=)
2. Open console so you can see submit data when you click submit button.
3. Enable Omit extra data checkbox
4. Add an additional property (click plus button) 
5. Click submit
6. Check submitted formData in console ... it is empty (why?)

#### Expected behavior

Additional Properties submitted in form data.

#### Actual behavior

Nothing submitted.

### Version

Used online playground version.

image

### PR DESCRIPTION
**Pull request by ValYouW**
### Reasons for making this change

Fix bug where additional properties are not submitted when nested and omit data enabled 
Fixes #1497 

### Checklist

* [ NA] **I'm updating documentation**
  - [ ] I've [checked the rendering](https://react-jsonschema-form.readthedocs.io/en/latest/#contributing) of the Markdown text I've added
* [X ] **I'm adding or updating code**
  - [X ] I've added and/or updated tests
  - [NA ] I've updated [docs](https://react-jsonschema-form.readthedocs.io/) if needed
* [ NA] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

### CODE REVIEW

**Comment by ValYouW** [comment]
A suggested simple fix for the issue, changes seem contained but need a pair of experienced eyes...
Thx.

**Comment by ValYouW** [comment]
If anyone is interested in trying this fix I published the `core` package to the [GitHub registry](https://github.com/ValYouW/react-jsonschema-form/packages/245524).
For instructions on installing packages from GitHub see [this](https://help.github.com/en/packages/using-github-packages-with-your-projects-ecosystem/configuring-npm-for-use-with-github-packages#installing-a-package)

**Comment by achendrick** [comment]
@ValYouW I just tested out my project that had this issue and it resolves it. I also ran a few other tests on already working forms and had no issues. Thanks for tackling this!

**Comment by ValYouW** [comment]
Hi @epicfaace can this be merged? anything I can do to help?

File: packages/core/src/components/Form.js [hunk 1]
```diff
@@ -196,7 +196,12 @@ export default class Form extends Component {
       Object.keys(_obj).forEach(key => {
         if (typeof _obj[key] === "object") {
           let newPaths = paths.map(path => `${path}.${key}`);
-          getAllPaths(_obj[key], acc, newPaths);
+          // if object marked with additionalProperties all its keys are valid
```
Comment (epicfaace) [inline packages/core/src/components/Form.js:199]:
```suggestion
          // If an object is marked with additionalProperties, all its keys are valid
```

File: packages/core/test/Form_test.js [hunk 2]
```diff
@@ -2808,6 +2808,53 @@ describe("Form omitExtraData and liveOmit", () => {
...
+
+      const pathSchema = {
+        $name: "",
+        level1: {
+          $name: "level1",
+          level2: { $name: "level1.level2" },
+          mixedMap: {
+            $name: "level1.mixedMap",
+            $additionalProperties: true,
+            namedField: { $name: "level1.mixedMap.namedField" }, // this should not be returned as additionalProperties include all
```
Comment (epicfaace) [inline packages/core/test/Form_test.js:2844]:
```suggestion
            namedField: { $name: "level1.mixedMap.namedField" }, // this name should not be returned, as the root object paths should be returned for objects marked with additionalProperties
```

File: packages/core/test/Form_test.js [hunk 3]
```diff
@@ -2966,9 +3013,13 @@ describe("Form omitExtraData and liveOmit", () => {
       properties: {
         foo: { type: "string" },
         bar: { type: "string" },
+        add: {
```
Comment (epicfaace) [inline packages/core/test/Form_test.js:3016]: Can you actually keep this test the same, then make another test that specifically tests "should not omit additionalProperties on change with omitExtraData=true and liveOmit=true"?

File: packages/core/src/utils.js [hunk 4]
```diff
@@ -990,6 +990,11 @@ export function toPathSchema(schema, name = "", rootSchema, formData = {}) {
     const _schema = retrieveSchema(schema, rootSchema, formData);
     return toPathSchema(_schema, name, rootSchema, formData);
   }
+
+  if (schema.hasOwnProperty("additionalProperties")) {
+    pathSchema.$additionalProperties = true;
```
Comment (epicfaace) [inline packages/core/src/utils.js:995]: Can you rename `$additionalProperties` to `__rjsf_additionalProperties`?

I know we haven't been namespacing special properties like this before, but I think we should start doing it from now on whenever we add new ones.

**Comment by epicfaace** [comment]
Can you also add this test (https://github.com/rjsf-team/react-jsonschema-form/issues/1497#issuecomment-628226728) in your PR?

**Comment by ValYouW** [comment]
@epicfaace done.
As for the extra test you wrote I had to change
```js
const textNode = node.querySelector("#root_key1-key");
```
to
```js
const textNode = node.querySelector("#root-key");
```

**Comment by achendrick** [comment]
Thanks for the work on this @ValYouW and @epicfaace 👍

**Pull request merged**